libraries/node-core-library/src/Sort.ts

```diff
@@ -231,4 +231,92 @@ export class Sort {
       set.add(item);
     }
   }
+
+  /**
+   * Sort the keys given in an object
+   *
+   * @example
+   *
+   * ```ts
+   * console.log(Sort.sortKeys({ c: 3, b: 2, a: 1 })); // { a: 1, b: 2, c: 3}
+   * ```
+   */
+  public static sortKeys<T extends Partial<Record<string, unknown>> | unknown[]>(
+    object: T,
+    { deep, compare }: { deep?: boolean; compare?: (x: string, y: string) => number } = {
+      deep: false,
+      compare: Sort.compareByValue
+    }
+  ): T {
+    function isPlainObject(obj: unknown): obj is object {
+      return obj !== null && typeof obj === 'object';
+    }
+    if (!isPlainObject(object) && !Array.isArray(object)) {
+      throw new TypeError(`Expected object or array`);
+    }
+
+    const cache: WeakMap<
+      Partial<Record<string, unknown>> | unknown[],
+      Partial<Record<string, unknown>> | unknown[]
+    > = new WeakMap();
+
+    function innerSortArray(arr: unknown[]): unknown[] {
```

Comment:
Nit: move these inline functions out of the class and into the module level. Makes them easier to read and ensures that the function is only defined once, when the module is loaded

Reply:
Updated